### DISCUSSION TITLE
Different Results in plasticity models

### DISCUSSION THREAD

**Discussion by walden37** [idaholab/moose#23240] [Q&A Modules: Solid mechanics] [answered]
Hi, 
I have recently been having problems with a thermal mechanical simulation with plasticity. I have implemented plasticity in the old tensor mechanics system through use of the TemperatureDependentHardeningStressUpdate material and in the new system with the ComputeSimoHughesJ2PlasticityStress material.  The stresses and strains in the cube differ greatly (approximately ten orders of magnitude) for the two models, despite the eigenstrains almost matching.

The model set up is as follows:
A cube of length 200mm made of titanium 
Left face heated to a temperature of 2000K 
Right face heated to a temperature of 300K
Left face of the cubes displacement is fixed in x, y and z 
The cube has an initial temperature and stress free reference temperature of 1100K
The stresses and strains are measured at:
measurePoint0=5
measurePoint1=12.5 
measurePoint2=25 
I have curve fit the PiecewiseLinear hardening functions to use for the Simo-Hughes model.
I have attached the input files that produce the different results.

I would expect the stresses and strains to approximately match. Could this be a misunderstanding of the Simo-Hughes plasticity model or possibly be a bug?

Many Thanks 
[SimoHughes.txt](https://github.com/idaholab/moose/files/10492270/SimoHughes.txt)
[Isotropic.txt](https://github.com/idaholab/moose/files/10492276/Isotropic.txt)

**Comment by hugary1995** [comment, marked as answer]
Thanks for reporting. I'll look into this.

**Reply by hugary1995** [reply]
OK, so I compared the two material models (one setup with the old system, the other with the new system) at two different temperatures (T = 300 & 2000). The parameters are the same as yours. I am testing it on a single-element cube, and I am not solving for the temperature -- the temperature remains homogeneous, hence the eigenstrain.

See the figure below. The stress magnitudes are the same.

![Figure_1](https://user-images.githubusercontent.com/31555204/214635149-db217849-1e7e-4bfa-8e95-63d7544a123d.png)

So the conclusion is the material model with the same hardening function has similar performance, regardless of which TM system you use. 

The difference you see in your problem is probably due to different localization phenomena in the old and the new systems, as the formulations are different. The old system uses hypoelasticity, whereas the specific Simo-Hughes material in the new system uses hyperelasticity. Again, there is no single answer as to which formulation is better.

The input files I used can be found on this branch: https://github.com/hugary1995/moose/tree/old_new_TM_isotropic_plasticity_comparison

**Reply by hugary1995** [reply]
uhh, I see the problem here -- the Simo-Hughes material isn't accounting for the eigenstrain for some reason...

**Reply by hugary1995** [reply]
Converting this to an issue...

**Reply by hugary1995** [reply]
Here: #23245
